jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/HelpfulSuggestions.java

```diff
@@ -193,27 +172,37 @@ public String forCacheDirectoryNotOwned(Path cacheDirectory) {
   }
 
   public String forHttpStatusCodeForbidden(String imageReference) {
-    return suggest("make sure you have permissions for " + imageReference);
+    return suggest(
+        "make sure you have permissions for "
+            + imageReference
+            + " and set correct credentials. See "
+            + "https://github.com/GoogleContainerTools/jib/blob/master/docs/faq.md#what-should-i-do-when-the-registry-responds-with-forbidden-or-denied for help");
   }
 
   public String forNoCredentialsDefined(String registry, String repository) {
     Preconditions.checkNotNull(baseImageReference);
     Preconditions.checkNotNull(targetImageReference);
-    if (noCredentialsDefinedForBaseImage
-        && registry.equals(baseImageReference.getRegistry())
-        && repository.equals(baseImageReference.getRepository())) {
-      return forNoCredentialHelpersDefined(
-          baseImageCredHelperConfiguration, baseImageAuthConfiguration.apply(registry));
-    }
-    if (noCredentialsDefinedForTargetImage
-        && registry.equals(targetImageReference.getRegistry())
-        && repository.equals(targetImageReference.getRepository())) {
-      return forNoCredentialHelpersDefined(
-          targetImageCredHelperConfiguration, targetImageAuthConfiguration.apply(registry));
+
+    final String unauthorizedEntity;
```

Comment:
Maybe we can just do `String unauthorizedEntity = registry;` and remove the else clause so we can declare and initalize on the same line.